tests/scripts/test.js

```diff
@@ -0,0 +1,829 @@
+/**
+ * Copyright (c) 2015-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License.
+ * Please see the LICENSE included with this distribution for details.
+ */
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+const colors = require('colors'); // eslint-disable-line no-unused-vars
+const ejs = require('ejs');
+const StreamSplitter = require('stream-splitter');
+const spawn = require('child_process').spawn; // eslint-disable-line security/detect-child-process
+const titanium = require.resolve('titanium');
+const { callbackify, promisify } = require('util');
+const exec = promisify(require('child_process').exec); // eslint-disable-line security/detect-child-process
+const SOURCE_DIR = path.join(__dirname, '..');
+const PROJECT_NAME = 'mocha';
+const APP_ID = 'com.appcelerator.testApp.testing';
+const PROJECT_DIR = path.join(__dirname, PROJECT_NAME);
+const JUNIT_TEMPLATE = path.join(__dirname, 'junit.xml.ejs');
+
+async function clearPreviousApp() {
+	// If the project already exists, wipe it
+	if (await fs.exists(PROJECT_DIR)) {
+		return fs.remove(PROJECT_DIR);
+	}
+}
+
+/**
+ * @param {string} sdkDir filepath to sdk install directory
+ * @param {string} sdkVersion version to install
+ * @returns {Promise<void>}
+ */
+async function installSDK(sdkDir, sdkVersion) {
+	const args = [ titanium, 'sdk', 'install', '--no-banner', '--no-prompt' ];
+	if (sdkVersion.indexOf('.') === -1) { // no period, probably mean a branch
+		args.push('-b');
+	}
+	args.push(sdkVersion);
+	args.push('-d'); // make default
+	// Add force flag if we find that the modules dir is blown away!
+	// eslint-disable-next-line promise/always-return
+	if (!(await fs.exists(path.join(sdkDir, 'modules')))) {
+		args.push('--force'); // make default
+	}
+	if (process.env.JENKINS || process.env.JENKINS_URL) {
+		args.push('--no-progress-bars');
+	}
+
+	console.log(`Installing SDK with args: ${args.join(' ')}`);
+	return new Promise((resolve, reject) => {
+		const prc = spawn('node', args, { stdio: 'inherit' });
+		prc.on('exit', code => {
+			if (code !== 0) {
+				return reject(new Error('Failed to install SDK'));
+			}
+			resolve();
+		});
+	});
+}
+
+/**
+ * Look up the full path to the SDK we just installed (the SDK we'll be hacking
+ * to add our locally built Windows SDK into).
+ *
+ * @returns {string}
+ **/
+async function getSDKInstallDir() {
+	// TODO: Use fork since we're spawning off another node process
+	const { stdout, _stderr } = await exec(`node "${titanium}" info -o json -t titanium`);
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/scripts/tes...
  
  
  DangerID: danger-id-Danger;
  File: tests/scripts/test.js;
  Line: 71;
-->

- :warning: tests/scripts/test.js line 71 – '_stderr' is assigned a value but never used. (no-unused-vars)